specs/~dynamic-names.yml

```diff
@@ -0,0 +1,151 @@
+    partials: { missing: 'Hello, world!' }
+    expected: '""'
+
+  - name: Failed Lookup - Partial
+    desc: The empty string should be used when the named partial is not found.
+    data: { dynamic: 'content' }
+    template: '"{{>*dynamic}}"'
+    partials: { foobar: 'Hello, world!' }
+    expected: '""'
+
+  - name: Context
+    desc: The dynamic partial should operate within the current context.
+    data: { text: 'Hello, world!', example: 'partial'  }
+    template: '"{{>*example}}"'
+    partials: { partial: '*{{text}}*' }
+    expected: '"*Hello, world!*"'
+
+  - name: Recursion
+    desc: Dynamic partials should properly recurse.
+    data:
+      template: 'node'
+      content: 'X'
+      nodes: [ { content: 'Y', nodes: [] } ]
+    template: '{{>*template}}'
+    partials: { node: '{{content}}<{{#nodes}}{{>*template}}{{/nodes}}>' }
+    expected: 'X<Y<>>'
+
+  # Whitespace Sensitivity
+
+  - name: Surrounding Whitespace
+    desc: |
+      A dynamic partials should not alter surrounding whitespace; any
+      whitespace preceding the tag should treated as indentation while any
+      whitepsace succeding the tag should be left untouched.
+    data: { partial: 'foobar' }
+    template: '| {{>*partial}} |'
+    partials: { foobar: "\t|\t" }
+    expected: "| \t|\t |"
+
+  - name: Inline Indentation
+    desc: |
+      Whitespace should be left untouched: whitespaces preceding the tag
+      should be treated as indentation.
+    data: { dynamic: 'partial', data: '|' }
+    template: "  {{data}}  {{>* dynamic}}\n"
+    partials: { partial: ">\n>" }
+    expected: "  |  >\n>\n"
+
+  - name: Standalone Line Endings
+    desc: '"\r\n" should be considered a newline for standalone tags.'
+    data: { dynamic: 'partial' }
+    template: "|\r\n{{>*dynamic}}\r\n|"
+    partials: { partial: ">" }
+    expected: "|\r\n>|"
+
+  - name: Standalone Without Previous Line
+    desc: Standalone tags should not require a newline to precede them.
+    data: { dynamic: 'partial' }
+    template: "  {{>*dynamic}}\n>"
+    partials: { partial: ">\n>"}
+    expected: "  >\n  >>"
+
+  - name: Standalone Without Newline
+    desc: Standalone tags should not require a newline to follow them.
+    data: { dynamic: 'partial' }
+    template: ">\n  {{>*dynamic}}"
+    partials: { partial: ">\n>" }
+    expected: ">\n  >\n  >"
+
+  - name: Standalone Indentation
+    desc: Each line of the partial should be indented before rendering.
+    data: { dynamic: 'partial', content: "<\n->" }
+    template: |
+      \
+       {{>*dynamic}}
+      /
+    partials:
+      partial: |
+        |
+        {{{content}}}
+        |
+    expected: |
+      \
+       |
+       <
+      ->
+       |
+      /
+
+  # Whitespace Insensitivity
+
+  - name: Padding Whitespace
+    desc: Superfluous in-tag whitespace should be ignored.
+    data: { dynamic: 'partial', boolean: true }
+    template: "|{{> *dynamic }}|"
+    partials: { partial: "[]" }
+    expected: '|[]|'
```

Comment:
If we believe that it is reasonable to have arbitrary whitespace between the `*` and the dotted name, *and* we believe this is important enough to spec it, then this would probably be the right place to insert another space. 

I actually think I'm against this, because spaces within dotted names are not specced, either. Still mentioning it here for completeness.

```suggestion
  - name: Padding Whitespace
    desc: Superfluous in-tag whitespace should be ignored.
    data: { dynamic: 'partial', boolean: true }
    template: "|{{> * dynamic }}|"
    partials: { partial: "[]" }
    expected: '|[]|'
```

(Ironically, there was a space in that exact position in your original proposal. I moved it one position to the left in my previous review. I think the space directly after the sigil is more important to spec, for consistency with the other specs, so this is why this third version has spaces in both places.)

Reply:
IMO we should support spaces in both places, by analogy to the pointer/dereference operator in c-like languages. You could read `{{>*dynamic}}` as `{{> *dynamic}}` or `{{>* dynamic}}`. We do this sort of thing with whitespace elsewhere, so it feels natural to extend to (and spec) this behavior here as well.